allocator.go

```diff
@@ -106,6 +115,18 @@ func WithTagMatching() Option {
 	}
 }
 
+func WithChurnConstraint(prevAssignment map[RangeID][]NodeID, minimizeToggle bool, maxChurn int64) Option {
+	return func(opt *options) {
+		opt.churnOptions.prevAssignment = prevAssignment
+		opt.churnOptions.WithMinimalChurn = minimizeToggle
+		opt.churnOptions.withMaxChurn = maxChurn < math.MaxInt64
```

Comment:
if maxChurn == inf (math.MaxInt64) then it would not apply maxChurn

Reply:
Let's see what this looks like once we consolidate Max/Min churn together, this one seems like it can be simplified.

Reply:
I think this whole thing can be more simplified if we implicitly try to balance instead of asking callers to have it specified. Having balancing exposed makes sense if they're to pass in some custom comparator/heuristic that we can use to control our spread, but if it's going to be a simplistic approach, might as well do that implicitly and have this be only for `maxChurn`. Thoughts @AlirezaDX @irfansharif @musanimu ?

Reply:
Agreed if we are not planning to expand on it , A simple approach would be sufficient

Reply:
Sorry I am not sure I follow, do you mean there is no need to specify max churn?

I think we should keep both around to be able to test how minimizing vs max churn constraint vs both would affect the performance.

Reply:
Actually never-mind, we can't implicitly balance since we need the previous allocation, which we're banking on the caller to have cached. Can we make the difference between minimizing churn and setting a hard limit more explicit by the naming? Also don't think we need to do :

`opt.churnOptions.withMaxChurn = maxChurn < math.MaxInt64`

If the caller asks for maxChurn with that value, why reject if it's at `math.MaxInt64`? If they don't want to use maxChurn with a limit set, they have the option of simply not passing in that option.

Reply:
Are you suggesting to have to methods that provide the options for Churn? One that takes a `MaxChurn `value and one that does not?

As far as I understood, Irfan suggested that we avoid creating multiple functions that do the same thing and instead similar to the OR tool wrapper utilize values such as `MaxInt` to do the above. I don't mind though to add 2 methods.

Reply:
I believe in another comment Irfan mentioned to use `MaxChurn = 0` as the flag for no `MaxChurn` constraint. I would argue that `MaxInt` is a more natural choice and less likely for the user of the client to use by mistake.

Reply:
I think it's better to not have a flag for maxChurn and simply use the value, we can set it to be -1 initially, and then once someone says `withMaxChurnSet(x)`, we can check if `x >= 0`, and assign it to maxChurn. Then when we want to constrain maxChurn, we check if maxChurn is unset and if so, don't apply the constraint. With maxInt it get's very obscure, i.e currently if I do : 
` allocator.withMaxChurn(maxInt64)`, it's the same as not calling `withMaxChurn` at all. Why give users an option that with a specific argument renders the whole call obsolete? I think we should keep it as simple as possible, have a value for maxChurn initialized to some constant, if a user calls `withMaxChurn`, sanitize the arg, set maxChurn equal to it, and apply the constraint.

Reply:
What happens if the user only uses `withMaxChurnSet(x)` without providing the previous assignment?